Data/Graph/Inductive/Query/SP/DAG.hs

```diff
@@ -0,0 +1,200 @@
+-- Module written by Hunter Herman
+-- Contact @ hherman1@macalester.edu
+-- | An implementation of O(N + E) algorithms for finding shortest
+--   and longest paths on directed acyclic graphs. Most functions
+--   require the input of a topological sort of the graph and 
+--   the length of the topological sort. A topological sort can
+--   be obtained with the 'topsort' function in Data.Graph.Inductive.Query.DFS
+--
+--   It is important to note that the library does not check if
+--   the input graph is a DAG. Unreliable results will be returned on 
+--   any other graph. 
+--   
+--   It may be of interest to some that it is not necessary to output a list
+--   of Nodes from these functions. The shortest/longest path functions are simple
+--   combinations of runPaths, tracePath, and Data.Vector.maxIndex. 
+--
+--   The algorithms are adaptations of the following:
+--   <https://en.wikipedia.org/wiki/Topological_sorting#Application_to_shortest_path_finding>
+module Data.Graph.Inductive.Query.SP.DAG (
+      shortestPathBetweenNodes
+    , longestPathBetweenNodes
+    , longestPathFromSource
+    , longestPath
+    , tracePath
+    , runPaths
+    , Number
+) where
+
+import Prelude hiding (read)
+import Data.Graph.Inductive.Graph
+import Data.Vector (Vector,maxIndex,freeze,(!))
+import Data.Vector.Mutable (IOVector,write,set,read,new)
+import Control.Monad (MonadPlus,mplus,mzero)
+import Data.Foldable (foldl',Foldable)
+import Data.Functor ((<$>))
+-- | A wrapper for any number, including Infinity and NegativeInfinity
+--   options.
+data Number a = Infinity | NegativeInfinity | Number a deriving (Show, Eq)
```

Comment:
Please at least put the `deriving` on a new line (also, any particular reason for not deriving `Read`?).